nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -312,100 +342,236 @@ func (w *deploymentWatcher) StopWatch() {
 	w.exitFn()
 }
 
-// watch is the long running watcher that takes actions upon allocation changes
+// watch is the long running watcher that watches for both allocation and
+// deployment changes. Its function is to create evaluations to trigger the
+// scheduler when more progress can be made, to fail the deployment if it has
+// failed and potentially rolling back the job.
 func (w *deploymentWatcher) watch() {
+	// Get the deadline. This is likely a zero time to begin with but we need to
```

Comment:
Maybe something like "The deadline, upon initialization, is zero. If we are slow to start watching it, the deployment will progress and the deadline will increase." Or something that describes why the deployment is initially zero but occasionally not.

Reply:
Added an example to when it would occur